cmake/kokkos_arch.cmake

```diff
@@ -622,6 +622,9 @@ IF (KOKKOS_ENABLE_SYCL)
   COMPILER_SPECIFIC_OPTIONS(
     DEFAULT -fsycl-unnamed-lambda
   )
+  IF (NOT KOKKOS_ENABLE_IMPL_SYCL_RELOCATABLE_DEVICE_CODE)
+    COMPILER_SPECIFIC_OPTIONS(DEFAULT -fno-sycl-rdc)
+  ENDIF()
```

Comment:
What made you change your mind on explicitly passing `-fsycl-rdc` otherwise?
Not asking you to change, just curious.

Reply:
oneAPI 2023.0.0 doesn't recognize -f[no-]sycl-rdc (as stated in the commit message). So disabling only works for newer compiler versions (2023.1.0 would be sufficient).